### ISSUE TITLE
RFC: drop support for beamer notes

### ISSUE THREAD

**Issue by fnevgeny** [pdfpc/pdfpc#472]
The current support for the beamer notes (`\setbeameroption{show notes on second screen}`) is inconsistent with the way "normal" notes (either manually entered and stored in .pdfpc, or PDF annotations) are handled and, moreover, add very little value compared to what can be achieved using the native PDF annotations (see #471). Except for visual nicety, the only functionality currently missing is per-overlay notes (#239), but it can be implemented quite easily.

So I suggest to drop it entirely. This will also result in a cleaner code with less bugs, which pop up once in a while (#465, #353, #454, #416, #435, #397, #280,...). Some of those issues are "feature requests" but IMO they are actually asking to fix or work around inconsistencies in the way beamer notes are currently displayed.

**Comment by AndreasBilke** [comment]
Apart from the fact that latex beamer notes was my first feature I
implemented in pdfpc, I agree with your points :)

Dropping the support should make the code more readable, less buggy and the
behaviour more predictable (e.g. disable auto grouping if notes are used).

**Comment by einhander** [comment]
Hello, please don't drop beamer notes, it's a main feature of pdfpc.
None of other methods don't support latex features like hiding text, lists, tables, colors etc.

**Comment by fnevgeny** [comment]
> None of other methods don't support latex features like hiding text, lists, tables, colors etc.

@einhander are you talking about *notes*??

**Comment by einhander** [comment]
Yes, \note command support almost all latex feature for text, even images and tikz graphics.
Pdf annotations and .pdfpc  plain text notes doesn't have such abilities.

**Comment by einhander** [comment]
Here some example of customized notes page with images table and color text:
![123](https://user-images.githubusercontent.com/5439042/70387984-fac54600-19bc-11ea-945e-f41cad0e6803.jpg)

**Comment by fnevgeny** [comment]
Well, I am genuinely intrigued - not by the possibility to make such complex notes, but why would someone need them. Can you please elaborate a bit? After all, notes are only for, and seen only by, you (the presenter); why spend time on bells and whistles while preparing them?

**Comment by einhander** [comment]
Generally, you are right about bells and whistles, but I frequently use itemize, math and chemical formulas. 
Also \note command is standard way to do a handouts. I use pdfpc on linux and other software on windows xp, so I'll have to do a double work to create a presentation.

**Comment by fnevgeny** [comment]
OK, I see. But then I'm pretty sure you don't use the "normal" pdfpc notes, right? That is to say, their widget still occupies its screen space with no use. Wouldn't it be better to render the beamer notes instead in the same place and keep the rest of the logics (the "next" slide etc) as usual? All this as a run-time option, of course.

**Comment by einhander** [comment]
> Wouldn't it be better to render the beamer notes instead in the same place and keep the rest of the logics (the "next" slide etc) as usual? All this as a run-time option, of course.

This would be great👍🏻

**Comment by gkuenning** [comment]
For me, the advantage of Beamer notes is that everything is in one place.  In other words, when I'm editing the presentation I can add notes right there, where they're next to the slide.  It would be a huge PITA to keep them in a different file (and in a different format) and have to keep the two files in sync.

If the PDFPC notes could be extracted from the Beamer source, even as a separate step that I could automate in a Makefile, then I'm agnostic.

**Comment by fnevgeny** [comment]
> For me, the advantage of Beamer notes is that everything is in one place.

Which is exactly the idea of #300/#471.

**Comment by AndreasBilke** [comment]
So beamer note support will be kept in pdfpc?

**Comment by fnevgeny** [comment]
@AndreasBilke Well, I _really_ don't like beamer notes. IMO, it was a hack in the first place (I'm talking about beamer itself), allowing to utilize a two-monitor setup without a specialized application like pdfpc, and a buggy one (the bug in pgfpages); and this bug breaks some nice features of pdfpc. On the other hand, there is a clear demand from some users for advanced typesetting in notes beyond the plain text.

So which options do we have?

**Comment by AndreasBilke** [comment]
> So which options do we have?

Drop it and fear the wrath of the users or keep and live with it :)

If we keep it then the markdown support for notes issue can be closed.
Otherwise there is a good point on it for having more advanced notes
features.

I would hand over the decision to you since you are way more active in this
project currently.

**Comment by fnevgeny** [comment]
I suggest to postpone the final (re)solution until after 4.4 and keep the issues open meantime... It won't be a small project either way...

**Comment by fnevgeny** [comment]
Actually, after looking more deeply into markdown, I tend more favorably towards it (I hardly used it before except writing some basic readme's). So it's quite powerful by itself, but can also include html, and so rendering chemical & math formulae etc shouldn't be an issue.

It's still a pity to use anything else when TeX can do it already, but given multiple problems with the beamer notes on each side of the implementation, perhaps it's worth reconsidering....

**Comment by zaphod92** [comment]
Hello, I share einhander's view. Dropping the support for beamer notes would break my entire workflow... It is absolutely lovely to be able to use math, chemical formulas, SI units, abbreviations and macros in your notes directly from you main Latex file. 

I understand that the bugs inherited from pgfpages and beamer are out of your reach and will probably never be fixed, which is sort of unsatisfactory. But please accept these few passages of hacky code and slight inconsistencies for the sake of true Latex notes. Neither markdown nor the perfect pdfpc-Latex package can accomplish what a simple "\setbeameroption{show notes on second screen}" does. :(

**Comment by fnevgeny** [comment]
> Neither markdown nor the perfect pdfpc-Latex package can accomplish what a simple "\setbeameroption{show notes on second screen}" does.

Why (except the obvious need to use a different markup language for notes)?

**Comment by zaphod92** [comment]
That is the reason. Using the same markup language for the entire presentation is beautiful and saves time. If this feature is dropped in 4.5, me and probably some other stubborn users will have to stick with 4.4, which would be sad for all of us.

**Comment by fnevgeny** [comment]
Perhaps the said stubborn users (supposedly highly versed in LaTeX, BTW) might want to fix the respective beamer/pgf bugs or at least submit/confirm/reinforce bug reports/nudge the maintainers to fix them/etc **there**? Just a thought...

**Comment by pank** [comment]
I have to do some online lectures from my laptop with a single display. `pdfpc` is pretty good for "screencasting" lectures as I can have a separate display window and presenter windows and use OBS to capture the display window from another virtual desktop.

However, at this point I do not enjoy using the `pdfpc.sty` LaTeX package at all, whereas even in its current state, I can at least produce beamer notes.

Example of annoyances; 
1. minor: e.g. `duration` isn't working on the CTAN version (`v0.2.1`) of `pdfpc.sty`, although it is working in the github version (`v0.3`).
2. the LaTeX `pdfcomments` package seems very fickly. E.g. `\par` seems to cause an error. The errors messages are not easy to debug. This is both using `xelatex` and `pdflatex`. 

Thus, I would personally much prefer #479.

I'm sympathetic to the view that any errors should be fixed in `beamer`, but perhaps it is still in better state than `pdfcomments.sty`?

Example re. point 2: 
```latex
\documentclass{beamer}
\usepackage[duration=45]{pdfpc}
\begin{document}
\begin{frame}{Title}
  foo
  \pdfpcnote{bar

    foobar}
\end{frame}  
\end{document}
```

log examle:
```
./test.tex:9: Paragraph ended before \HyPsd@@ProtectSpacesFi was complete.
<to be read again> 
                   \par 
l.9 \end{frame}
```

**Comment by AndreasBilke** [comment]
Could you please open a seperate issue for this?

**Comment by fnevgeny** [comment]
>     1. minor: e.g. `duration` isn't working on the CTAN version (`v0.2.1`) of `pdfpc.sty`, although it is working in the github version (`v0.3`).

So this item irrelevant.

>     2. the LaTeX `pdfcomments` package seems very fickly. E.g. `\par` seems to cause an error. 

Yes, the comments are limited to one paragraph, it's mentioned in the `pdfcomments` docs. You can use two (or more) separate comments.

**Comment by fnevgeny** [comment]
@pank see #535, `\\` is supported as a line break.

**Comment by fnevgeny** [comment]
Closing for now...

**Comment by MartinX3** [comment]
Sadly the example demo presentation has no slide notes to test with this pdf viewer :(

**Comment by fnevgeny** [comment]
> Sadly the example demo presentation has no slide notes to test with this pdf viewer :(

Press `Ctrl+N` and begin typing your own :). Or do you mean notes of the beamer types? These break the autogrouping feature (please read this discussion from the beginning) and that is considered much more important feature than beamer notes. So obviously, the standard demo will never have them.

**Comment by MartinX3** [comment]
> > Sadly the example demo presentation has no slide notes to test with this pdf viewer :(
> 
> Press `Ctrl+N` and begin typing your own :). Or do you mean notes of the beamer types? These break the autogrouping feature (please read this discussion from the beginning) and that is considered much more important feature than beamer notes. So obviously, the standard demo will never have them.

Ah, thank you!
Did I miss some readme? Because I don't find any `Ctrl+N` reference in this repo :)

And yeah, it is sad and I need to find a good solution.
dspdfviewer was great in combination with the `beamer notes on second screen` feature.
But I assume due to its missing development it will contain some security issues.

At the moment I see 2 solutions for my LaTeX presentations
- Learn the new syntax and add *.pdfpc files to my project
- Trying to use https://github.com/cebe/pdfpc-latex-notes which doesn't seem to be in the CTAN and isn't updated since 2010-02-01 but which will convert normal `\notes` into *.pdfpc files.

**Comment by fnevgeny** [comment]
> Did I miss some readme? Because I don't find any `Ctrl+N` reference in this repo :)

E.g., in the example demo you complained about :). Then the man page of course, or just press `?` while in the app...

> At the moment I see 2 solutions for my LaTeX presentations
> 
>     * Learn the new syntax and add *.pdfpc files to my project
> 
>     * Trying to use https://github.com/cebe/pdfpc-latex-notes which doesn't seem to be in the CTAN and isn't updated since 2010-02-01 but which will convert normal `\notes` into *.pdfpc files.

* https://ctan.org/pkg/pdfpc

**Comment by MartinX3** [comment]
> > Did I miss some readme? Because I don't find any `Ctrl+N` reference in this repo :)
> 
> E.g., in the example demo you complained about :). Then the man page of course, or just press `?` while in the app...
> 

Woopsi,thank you! I completely focused on this repo and its webpage. :)

> > At the moment I see 2 solutions for my LaTeX presentations
> > ```
> > * Learn the new syntax and add *.pdfpc files to my project
> > 
> > * Trying to use https://github.com/cebe/pdfpc-latex-notes which doesn't seem to be in the CTAN and isn't updated since 2010-02-01 but which will convert normal `\notes` into *.pdfpc files.
> > ```
> 
>     * [ctan.org/pkg/pdfpc](https://ctan.org/pkg/pdfpc)

Ah, I assume that package will do the same (converting \notes to the needed *.pdfpc files)! Thank you! :)

**Comment by einhander** [comment]
@MartinX3 did you tried the latest master on your pdfs with beamer notes on second screen feature? It works (@fnevgeny correct me if I'm wrong).

**Comment by MartinX3** [comment]
> @MartinX3 did you tried the latest master on your pdfs with beamer notes on second screen feature? It works (@fnevgeny correct me if I'm wrong).

I will try now the normal CTAN version, the entire topic discussion was a bit overwhelming with informations for me. :D
If it maybe doesn't work, I will report it. :D

**Comment by MartinX3** [comment]
Yes, you're right.
Only v0.6.0 has support for LuaLaTeX.

But i need to manually apply `--notes=right` if I use `\setbeameroption{show notes on second screen}` or `\setbeameroption{show notes on second screen = right}`.

Also it hides my `\pdfpcnote{bla}` stuff then.
But if i don't use the beamer option, my notes get rendered as I would write them with `CTRL+N` :)

But it runs amok, if I use ÄÖÜäöü in my notes. (I'm german)

**Comment by fnevgeny** [comment]
> Also it hides my `\pdfpcnote{bla}` stuff then.
> But if i don't use the beamer option, my notes get rendered as I would write them with `CTRL+N` :)

Unless you're completely sold on using *TeX typesetting to write your notes, there is really *no* need to use beamer notes...

> But it runs amok, if I use ÄÖÜäöü in my notes. (I'm german)

See #579.

In general, if you have a problem, open a new issue. Let's not overload the irrelevant and long ago closed topic.

**Comment by MartinX3** [comment]
Thank you!
I didn't search for "umlauts" since it sounds way too german. XD

**Comment by cpitclaudel** [comment]
> Unless you're completely sold on using *TeX typesetting to write your notes, there is really no need to use beamer notes...

Compatibility with other PDF viewers (like pympress) is another factor

**Comment by gkuenning** [comment]
As a general rule, it's painful and error-prone to try maintain 
one set of information that is spread across two different files. 
That's true whether it's keeping header files in sync with source 
code, or keeping notes in sync with the slides that they're about. 
The advantage of using in-presentation notes is that everything is 
in one place.  That's a huge win even if you don't need equations 
in your notes.

So yeah, I'm *completely* sold on using *TeX typesetting to write 
my notes.

>> Unless you're completely sold on using *TeX typesetting to 
>> write
> your notes, there is really no need to use beamer notes...
>
> Compatibility with other PDF viewers (like pympress) is another 
> factor
>
> -- 
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub:
> https://github.com/pdfpc/pdfpc/issues/472#issuecomment-796445298

-- 
    Geoff Kuenning   ***@***.*** 
    http://www.cs.hmc.edu/~geoff/

Orchestra retrospectively extremely satisfied with symphony 
[No. 1] as
result of barrel of free beer.
	-- Gustav Mahler, post-premiere letter to Arnold Berliner

**Comment by fnevgeny** [comment]
@gkuenning You're attacking a straw man. Did you actually read the first post in this "thread"? Have you tried using [pdfpc.sty](https://ctan.org/tex-archive/macros/latex/contrib/pdfpc)?

**Comment by MartinX3** [comment]
Thanks to the ctan package i can write my notes into the latex files like before without needing to create a second file for notes :)

**Comment by fnevgeny** [comment]
> Compatibility with other PDF viewers (like pympress) is another factor

@cpitclaudel I don't understand what you mean. Try persuading Ford they should make cars compatible with Nissan. Both should be "compatible" with the *standard* roads, that's all. But if one carmaker produces cars that happen to drive better on a dirt road in your backyard, fine - buy their cars.

**Comment by einhander** [comment]
@fnevgeny I think that @cpitclaudel talk about compatibilty with `\setbeameroption{show notes on second screen}` latex command. In term of car makers it's a _standard_ road right now. 
@cpitclaudel thanks for the hint with pympress, it's moved much further lately.
P.S. pdfpc is the best ;)

**Comment by gkuenning** [comment]
I read it several years ago when the thread was started.  I hope 
you'll forgive me for not recalling everything that has happened 
since.

> @gkuenning You're attacking a straw man. Did you actually read 
> the
> first post in this "thread"? Have you tried using
> [pdfpc.sty](https://ctan.org/tex-archive/macros/latex/contrib/pdfpc)?
>
> -- 
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub:
> https://github.com/pdfpc/pdfpc/issues/472#issuecomment-797409897

-- 
    Geoff Kuenning   ***@***.*** 
    http://www.cs.hmc.edu/~geoff/

McDonald's, which does not wait on your table, does not cook your 
food
to order, and does not clear your table, came up with the slogan 
``We
Do It All For You.''
	-- Dave Barry

**Comment by gkuenning** [comment]
Evgeny, one of the points of open-source software is that we try 
not to emulate the cutthroat competition of the commercial world. 
To the extent possible, we believe that it's best when we all get 
along.

Nobody expects Ford to make their cars compatible with Boeing 
airplanes, but we do indeed try to make Ford and Nissan cars 
compatible in many, many ways.  Some of those ways came about 
because of government regulation (e.g., headlight design) after 
the car companies proved they couldn't be trusted on their own. 
Others came about because it was good for the car companies 
themselves (e.g., standardized tire sizes).  But it is not a Good 
Thing that Ford and Nissan deliberately design minor parts (e.g., 
tire-pressure sensors) that are not cross-compatible.

In the open-source world, we try to be better than that.

>> Compatibility with other PDF viewers (like pympress) is another 
>> factor
>
> @cpitclaudel I don't understand what you mean. Try persuading 
> Ford
> they should make cars compatible with Nissan. Both should be
> "compatible" with the *standard* roads, that's all. But if one
> carmaker produces cars that happen to drive better on a dirt 
> road in
> your backyard, fine - buy their cars.
>
> -- 
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub:
> https://github.com/pdfpc/pdfpc/issues/472#issuecomment-797423324

-- 
    Geoff Kuenning   ***@***.*** 
    http://www.cs.hmc.edu/~geoff/

If at first you succeed, try to hide your astonishment.
	-- Harry F. Banks

**Comment by fnevgeny** [comment]
Well, we're too deep in the analogues... I don't believe that carmakers *deliberately* make parts not cross-compatible. And I don't think, either, that open source helps in this respect. Try "fitting" any "part" of pdfpc into pympress or vice versa. Possible, but requires quite some work. A good car technician will do the same with a tire sensor.

Anyhow, the only thing I want to say is that the "beamer notes" are in no way "standard". First, because there is no standards body that defined it. Second, there is exactly one "vendor" that makes them (so Ford or whoever made the tire sensor first might similarly claim their parts are "standard"). Third (and it's related to point 1) there are neither documentation on it nor sample implementation for interfacing with this feature (look at pdfpc.sty for a guesswork trying to detect the beamer notes even at the TeX compilation phase). Fourth, for years nobody cares of fixing an outstanding bug (which prevents pdfpc of enabling overlays with such presentations). Fifth (or actually it must the first) what kind of standard is it if 99% of existing PDF viewers wouldn't understand it? If you come to a conference and your laptop dies (or the organizers insist on using their PC) and you didn't care to produce also the no-notes variants, you are basically out of luck.

Compare this to using native PDF annotations which pdfpc supports (straightforwardly through pdfpc.sty in *TeX, but, in principle, using any PDF authoring software). PDF annotations are part of the PDF standard proper, and although most viewers will not allow you to see them during "slide show", the slides themselves will be rendered correctly. It's called graceful degradation (and lack of it in the case of beamer notes). Using Markdown, complex typesetting is possible, if needed. BTW, pympress is said to support native PDF annotations, too.

So what else except being "completely sold on using *TeX" makes one use the beamer notes nowadays??

**Comment by cpitclaudel** [comment]
> pympress is said to support native PDF annotations, too.

Is it?  I couldn't find a way to make that work.  I will look further.

> Third (and it's related to point 1) there are neither documentation on it

Really?  We should ask the beamer folks for help with that, then, because the rest of the documentation is excellent.  What is missing?

> nor sample implementation for interfacing with this feature

Fair, but the whole implementation of that feature is less than 200 lines of code (https://github.com/josephwright/beamer/blob/main/base/beamerbasenotes.sty)

> If you come to a conference and your laptop dies (or the organizers insist on using their PC) and you didn't care to produce also the no-notes variant, you are basically out of luck.

Completely right.  I wish there was a way to load two pdfs (one for the notes and one for the talk), but I don't have time to prepare a patch at the moment :/  Would that make things simpler on the pdfpc side?

> So what else except being "completely sold on using *TeX" makes one use the beamer notes nowadays??

I looked into pdfpc.sty.  Maybe it doesn't work with Beamer overlays, right? In pdfpc, the following example shows "note 1" on both slides:

```
\documentclass{beamer}

\usepackage{pdfpc}

\begin{document}
\begin{frame}
  \only<1>{
    Text 1
    \pdfpcnote{Note 1}
  }

  \only<2>{
    Text 2
    \pdfpcnote{Note 2}
  }
\end{frame}
\end{document}
```

The following works with Beamer's note pages:

```
\documentclass{beamer}

\usepackage{pdfpc}
\setbeameroption{show notes}

\begin{document}
\begin{frame}
  \only<1>{
    Text 1
    \note{Note 1}
  }

  \only<2>{
    Text 2
    \note{Note 2}
  }
\end{frame}
\end{document}
```

I don't think that counts as being completely sold on using TeX?

(The other use case for me is printing handouts with notes; for example I have a slide with a math result, I write the proof on the blackboard during class, and I include the proof in the beamer notes. Indeed, that counts as being sold on TeX ^^ but it's not my main use case)

**Comment by fnevgeny** [comment]
> Really? We should ask the beamer folks for help with that, then, because the rest of the documentation is excellent. What is missing?

A way to tell programmatically whether a given PDF file contains beamer notes (and if yes, on which side). Without involving AI.

Don't get me wrong. I don't blame beamer. But this way of rendering notes alongside the slide on the same page was a poor-man hack to imitate real dual-screen presentation software (which didn't exist at that time).

> Fair, but the whole implementation of that feature is less than 200 lines of code (https://github.com/josephwright/beamer/blob/main/base/beamerbasenotes.sty)

That's only part of it. But it's absolutely beyond the point. I don't have to look into code of something that produces something according to a *standard*. Here, there is no standard (or even attempt to make such).

> Completely right. I wish there was a way to load two pdfs (one for the notes and one for the talk), but I don't have time to prepare a patch at the moment :/ Would that make things simpler on the pdfpc side?

You mean putting more burden on pdfpc? No thanks :).

> I looked into pdfpc.sty. Maybe it doesn't work with Beamer overlays, right?

When you find something missing in pdfpc, there is a special place to submit request (hint: commenting on an issue closed long ago isn't it). pdfpc itself supports notes in overlays just fine. If you want to make it work with beamer, please go ahead. I suspect it will require modification of the beamer code.

> (The other use case for me is printing handouts with notes; for example I have a slide with a math result, I write the proof on the blackboard during class, and I include the proof in the beamer notes. Indeed, that counts as being sold on TeX ^^ but it's not my main use case)

OK, a fair reason.

BTW, libreoffice (when saving to pdf) makes the note slides *after* the slides proper, which is much closer to graceful degradation. It should be possible for pdfpc to support such an arrangement, but apparently nobody is interested. *Office folks use the same software for authoring and presentation :).

**Comment by cpitclaudel** [comment]
> A way to tell programmatically whether a given PDF file contains beamer notes (and if yes, on which side). Without involving AI.

Excellent point.  I have opened a feature request, asking the Beamer folks to add a PDF metadata field to indicate whether the document contains notes.  Of course, this will take time to propagate to clients once its implemented, but it's a step: https://github.com/josephwright/beamer/issues/669 .  If you have time, it would be great to confirm that this would work for PDFPC.

> Here, there is no standard (or even attempt to make such).

I understand.  But what would you want such a standard to say, except "the left half of the slide has the notes"?

> You mean putting more burden on pdfpc? No thanks :).

Oh, sorry, I thought (hoped?) this would make the implementation simpler.

> When you find something missing in pdfpc, there is a special place to submit request (hint: commenting on an issue closed long ago isn't it).

Yes, of course.  But I'm happy with the way pdfpc works currently with side-by-side notes, so there was no need to request extra functionality in a feature (pdfpc.sty) that I don't use. 

> pdfpc itself supports notes in overlays just fine. If you want to make it work with beamer, please go ahead. I suspect it will require modification of the beamer code.

I don't understand this part, sorry.  I'm very happy with the side by side notes support, so I haven't really looked at pdfpc.sty in depth; I was just answering the claim that it was straightforward to use native PDF annotations "through pdfpc.sty in *TeX".

> BTW, libreoffice (when saving to pdf) makes the note slides after the slides proper, which is much closer to graceful degradation.

I typically use `\setbeameroption{show notes on second screen=bottom}`, which does the same thing, doesn't it?

**Comment by fnevgeny** [comment]
>  but it's a step: [josephwright/beamer#669](https://github.com/josephwright/beamer/issues/669) . If you have time, it would be great to confirm that this would work for PDFPC.

Sure, thanks.

> I understand. But what would you want such a standard to say, except "the left half of the slide has the notes"?

Let's begin with this. Once the foundation is there, adding more properties should be trivial. And perhaps it's time to rethink the whole approach of placing notes (see below).

> I don't understand this part, sorry.

Open a "clean" presentation without notes and enter notes from pdfpc (`Ctrl-N`). You can do it on overlay slides. So internally, the mechanism is there. How to make it work using pdfpc.sty (or for that matter, use per-overlay any other TeX command of which beamer doesn't take intimate care of) I don't know.

> I typically use `\setbeameroption{show notes on second screen=bottom}`, which does the same thing, doesn't it?

Nope. There, first *all* real slides (1 .. N) go, followed by *all* note pages. Maybe this is what beamer should do as well, at least as an option.

**Comment by cpitclaudel** [comment]
> Sure, thanks.

Thanks a lot :)

> Open a "clean" presentation without notes and enter notes from pdfpc (Ctrl-N). You can do it on overlay slides. 

Got it.

> Nope. There, first all real slides (1 .. N) go, followed by all note pages. Maybe this is what beamer should do as well, at least as an option.

Ah!  I didn't realize.  Thanks for the pointer.  To display the notes along the document, I imagine we'd want per-page metadata to indicate which note page corresponds to which content page, right? (Is there already such metadata to indicate which pages are slides and which pages are new frames?  Or does pdfpc use special tricks to figure that out?)

But then, isn't that approach (all note pages at end of document) just a more complex version of the two-pdfs approach?

**Comment by fnevgeny** [comment]
> To display the notes along the document, I imagine we'd want per-page metadata to indicate which note page corresponds to which content page, right? (Is there already such metadata to indicate which pages are slides and which pages are new frames? Or does pdfpc use special tricks to figure that out?)

Guesswork based on page labels. Which is broken by a bug in `pgfpages` (#65).

> But then, isn't that approach (all note pages at end of document) just a more complex version of the two-pdfs approach?

No. It's better to deal with a single object. Not to mention preparing two PDF's (and ensuring they're in sync) is a pain for user...

**Comment by samcarter** [comment]
> Not to mention preparing two PDF's (and ensuring they're in sync) is a pain for user...

two in sync documents are actually rather easy

```
% !TeX program = pdflatex % | pdflatex -jobname=%_notes "\AtBeginDocument{\setbeameroption{show only notes}}\input{%}"| txs:///view-pdf "?am)_notes.pdf"
\documentclass{beamer}

\makeatletter 
\def\beamer@framenotesbegin{%	at beginning of slide 	
	\gdef\beamer@noteitems{}% 	
	\gdef\beamer@notes{{}}%	used to be totally	empty. 
}
\makeatother

\begin{document}

\begin{frame}
some content
\pause
some content
\note{note text}
\end{frame}

\begin{frame}
test
\end{frame}

\end{document}
```

(for editors which don't understand magic comments, compile with `pdflatex filename && pdflatex -jobname=%_notes "\AtBeginDocument{\setbeameroption{show only notes}}\input{filename}"`)

**Comment by fnevgeny** [comment]
Yes, it's possible of course. Also with makefile or like build system.

But I really would like to avoid adding support for yet another ad hoc notes approach.

**Comment by fnevgeny** [comment]
Re: beamer notes vs native PDF annotations as a "standard": I started a [list of PDF presenters at Wikipedia](https://en.wikipedia.org/wiki/Comparison_of_PDF_presentation_software). So far, it's roughly fifty-fifty in terms of support. And I guess adding commerical software will shift it towards PDF annotations... Feel free to contribute to the article, of course.

**Comment by gkuenning** [comment]
Sam, that's brilliant!  I didn't know you could put an 
AtBeginDocument on the command line.  It's going to make my life 
so much simpler.

>> Not to mention preparing two PDF's (and ensuring they're in 
>> sync) is
> a pain for user...
>
> two in sync documents are actually rather easy
>
> ```
> % !TeX program = pdflatex % | pdflatex -jobname=%_notes
> "\AtBeginDocument{\setbeameroption{show only notes}}\input{%}"|
> txs:///view-pdf "?am)_notes.pdf"
> \documentclass{beamer}
>
> \makeatletter 
> ***@***.***{%	at beginning of slide 	
> 	***@***.***{}% 	
> 	***@***.***{{}}%	used to be totally	empty. 
> }
> \makeatother
>
> \begin{document}
>
> \begin{frame}
> some content
> \pause
> some content
> \note{note text}
> \end{frame}
>
> \begin{frame}
> test
> \end{frame}
>
> \end{document}
> ```
>
> (for editors which don't understand magic comments, compile with
> `pdflatex filename && pdflatex -jobname=%_notes
> "\AtBeginDocument{\setbeameroption{show only
> notes}}\input{filename}"`)
>
> -- 
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub:
> https://github.com/pdfpc/pdfpc/issues/472#issuecomment-798786793

-- 
    Geoff Kuenning   ***@***.*** 
    http://www.cs.hmc.edu/~geoff/

The most exciting phrase to hear in science, the one that heralds 
new
discoveries, is not "Eureka!" (I found it!) but "That's funny ..."
                -- Isaac Asimov

**Comment by samcarter** [comment]
@gkuenning  If you wanted, you could have your whole document on the command line

```
pdflatex "\documentclass{article}\begin{document}hello world\end{document}"
```

Under the hood, calling `pdflatex filename` is basically the same as doing `pdflatex "\input{filename}"`

**Comment by michaelstats** [comment]
Can I please make a point in favour of keeping beamer native notes support in pdfpc? 

The brilliant emacs org mode provides a nice way of preparing a presentation using its lightweight markup, exporting to beamer. It also handles beamer notes. It is great being able to prepare a beamer presentation that way and having access to a pdf viewer which can take that and display it the way that pdfpc does (with timer, notes, etc. on the "second screen"). It would be a shame for pdfpc to drop support for that.
